creator-node/src/snapbackSM/snapbackSM.js

```diff
@@ -274,15 +291,112 @@ class SnapbackSM {
   }
 
   /**
-   * Submit POA:UserReplicaSetFactory.updateReplicaSet() transaction to move user off unhealthy node
+   * Depending on the size of `unhealthyReplicas`:
+   * 1. Determine a new replica set
+   * 2. Write new replica set to URSM
+   * 3. Sync data to new replica set
    *
-   * @dev TODO use libs for CreatorNodeSelection logic + relayed chain call
-   * @dev One issue - since it doesn't actually rectify broken replica sets, this will log every time it is processed
+   * @param {number} userId user id to issue a reconfiguration for
+   * @param {string} wallet wallet address of user id
+   * @param {string} primary endpoint of the current primary node on replica set
+   * @param {string} secondary1 endpoint of the current first secondary node on replica set
+   * @param {string} secondary2 endpoint of the current second secondary node on replica set
+   * @param {string[]} unhealthyReplicas array of endpoints of current replica set nodes that are unhealthy
+  */
+  async issueUpdateReplicaSetOp (userId, wallet, primary, secondary1, secondary2, unhealthyReplicas) {
+    this.log(`[issueUpdateReplicaSetOp] userId=${userId} wallet=${wallet} unhealthy replica set=[${unhealthyReplicas}]`)
+
+    const unhealthyReplicasSet = new Set(unhealthyReplicas)
+    const baseSyncRequestParams = { userWallet: wallet, syncType: SyncType.Manual }
+    let newReplicaSetSPIds = []
+
+    // If a primary is currently unhealthy or more than one node is unhealthy, skip for now
+    if (unhealthyReplicasSet.has(primary) || unhealthyReplicas.length > 1) {
+      this.log(`[issueUpdateReplicaSetOp] userId=${userId} skipping reconfig.`)
+      return
+    }
+
+    let phase = ''
+    try {
+      // Generate new replica set
+      phase = issueUpdateReplicaSetOpPhases.GET_HEALTHY_CONTENT_NODES
+      const currentHealthySecondary = unhealthyReplicasSet.has(secondary1) ? secondary2 : secondary1
+      const newSecondary = await this.determineNewReplicaSet({ wallet, secondary1, secondary2, primary })
+
+      this.log(`[issueUpdateReplicaSetOp] Updating userId=${userId} wallet=${wallet} replica set=[${primary},${secondary1},${secondary2}] to new replica set=[${primary},${currentHealthySecondary},${newSecondary}]`)
+
+      if (!this.snapbackReconfigEnabled) return
```

Comment:
this logic should be at the top of the function, no reason to do wasteful work.
also in this scenario we should still keep the log on 307

Reply:
will keep here to observe log behavior